File: taotao-parent/taotao-sellergoods-interface/src/main/java/com/taotao/sellergoods/service/BrandService.java
package com.taotao.sellergoods.service;

import com.taotao.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 */
public interface BrandService {

    /**
     * 删除品牌
     */
    void delete(Long[] ids);

    /**
     * 根据品牌ID获取实体
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌
     */
    void update(TbBrand brand);

    /**
     * 增加品牌
     */
    void add(TbBrand brand);

    /**
     * 条件分页列表
     */
    PageResult findPage(int pageNum, int pageSize,TbBrand brand);

    /**
     * 返回分页列表
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 返回全部列表
     */
    List<TbBrand> findAll();

    /**
     * 品牌下拉框数据
     */
    List<Map> selectOptionList();
}

File: taotao-parent/taotao-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService', function ($http) {
    //查询所有品牌
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    //分页查询
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };

    //查询实体
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    //新增品牌
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };

    //修改品牌
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };

    //删除品牌
    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    };

    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + "&rows=" + rows, searchEntity);
    };

    //品牌下拉列表数据
    this.selectOptionList = function(){
        return $http.get('../brand/selectOptionList.do');
    };
});